### PR TITLE
Add GitHub Actions and pre-commit scripts

### PR DESCRIPTION
**Pull request by Darvand**
## Pull request type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- No code coverage on GitHub checks and pre-commits

Issue Link: https://trello.com/c/IfxEqP8b/10-setup-code-coverage

## What is the new behavior?

- Add GitHub Actions and let them run scripts to check tests and code coverage. Add Husky library to execute the same scripts before a commit.

## Other information

N/A

## Preview

N/A

### CODE REVIEW

**Review by EMazzaglia — APPROVED** [review]
LGTM.

I've a question tho, is the code coverage threshold blocking the commit?

**Comment by Darvand** [comment]
```I've a question tho, is the code coverage threshold blocking the commit?```
Yes, it does

File: package.json [hunk 1]
```diff
@@ -41,6 +42,7 @@
     "eslint": "^8.0.1",
     "eslint-config-prettier": "^8.3.0",
```
Comment (slasia) [inline package.json:43]: I think you can move eslint, jest  and prettier to devDependencies, what do you think?

**Review by slasia — APPROVED** [review]
Good work! I just wrote a comment!

File: package.json (hunk 1 above)
Comment (Darvand) [reply package.json:43]: Yeah, but eslint, jest and prettier are already in the devDependencies

Comment (slasia) [reply package.json:43]: great!

**Review by slasia — APPROVED** [review]

**Pull request merged**